sklearn/tree/tests/test_export.py

```diff
@@ -311,6 +312,84 @@ def test_precision():
                              precision + 1)
 
 
+def test_export_text_errors():
+    clf = DecisionTreeClassifier(max_depth=2, random_state=0)
+    clf.fit(X, y)
+
+    assert_raise_message(ValueError,
+                         "max_depth bust be >= 0, given -1",
+                         export_text, clf, max_depth=-1)
+    assert_raise_message(ValueError,
+                         "feature_names must contain 2 elements, got 1",
+                         export_text, clf, feature_names=['a'])
+    assert_raise_message(ValueError,
+                         "decimals must be >= 0, given -1",
+                         export_text, clf, decimals=-1)
+    assert_raise_message(ValueError,
+                         "spacing must be > 0, given 0",
+                         export_text, clf, spacing=0)
+
+
+def test_export_text():
+    clf = DecisionTreeClassifier(max_depth=2, random_state=0)
+    clf.fit(X, y)
+
+    expected_report = dedent("""|--- feature_1 <= 0.00
```

Comment:
Not sure why you're using dedent but then not indenting all the lines. I might have expected:
```py
    expected_report = dedent("""
    |--- feature_1 <= 0.00
    |   |--- class: -1
    |--- feature_1 >  0.00
    |   |--- class: 1
    """).lstrip()
```

Reply:
Agree, now it looks like

```
    expected_report = dedent("""\
    |--- feature_1 <= 0.00
    |   |--- class: -1
    |--- feature_1 >  0.00
    |   |--- class: 1
    """)
```

It seems that there's no way to get rid of the ugly `\`

Reply:
Agree. Just modified accordingly.